cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/client/AzureClient.java

```diff
@@ -1021,4 +1022,17 @@ public void deleteDiskEncryptionSet(String resourceGroup, String diskEncryptionS
         return azureClientCredentials.getAccesToken();
     }
 
+    public boolean checkEncryptionKeyExistenceOnCloud(String vaultUri, String keyName, String keyVaultName, String keyVaultResourceGroupName) {
+        return handleAuthException(() -> {
+            PagedList<KeyItem> keys = azure.vaults()
+                    .getByResourceGroup(keyVaultResourceGroupName, keyVaultName)
+                    .client()
+                    .listKeys(vaultUri);
```

Comment:
Is my understanding right that the results are paginated and if the key is not present on the first page we will return `false`?

Reply:
Thank you for pointing this out. Fixing this.